### PR TITLE
Remove results tab when results not published yet

### PR DESCRIPTION
**Pull request by amelsmajic**
This fixes the issue #608.

I'm not sure if this is the correct way to do this. Any input @matthee?

### CODE REVIEW

File: app/views/exercises/_subnav.html.erb [hunk 1]
```diff
@@ -7,7 +7,7 @@
   <%= subnav_item(:widget, "Services", term_exercise_services_path(exercise.term, exercise), params[:controller] == "services")  if policy(ServicePolicy.term_policy_record(exercise.term)).index? %>
 
   <%= subnav_item(:list, "Submission", exercise_student_submission_path(exercise), params[:controller] == "submissions" || params[:controller] == "submission_tree") if current_account.student_of_term?(exercise.term) %>
-  <%= subnav_item(:megaphone, "Results", term_result_path(exercise.term, exercise), params[:controller] == "student_results") if policy(StudentResultsPolicy.term_policy_record(exercise.term)).index? %>
+  <%= subnav_item(:megaphone, "Results", term_result_path(exercise.term, exercise), params[:controller] == "student_results") if policy(StudentResultsPolicy.term_policy_record(exercise.term)).index? && GradingReview::SubmissionReview.find_by_account_and_exercise(current_account, exercise).published? %>
```
Comment (matthee) [inline app/views/exercises/_subnav.html.erb:10]: Have you tested this? I believe there will be an error raised when the student hasn't created a submission for this exercise. You might want to do this check differently. For example:

```ruby
Submission.active.for_account(current_account).for_exercise(exercise).first.try(:result_published?)
```

I would also like to see this functionality wrapped in a method, like `exercise.results_published_for_account?(current_account)`, but I'm open to your suggestions and API.

**Review by matthee — CHANGES_REQUESTED** [review]
Hi Amel,

thanks for your contribution!

There are three things I would like you to change: First of all, I think there is a sneaky bug, please refer to the inline code comment and make sure to fix it. Second, please extract this functionality into a method. And last but not least, please add tests, you might have catched that bug mentioned above on your own 😉

**Comment by amelsmajic** [comment]
I have now added tests and a wrapper in Exercise with the suggested code instead of my own.

The branch is already re-based on master and is ready to be merged.

**Comment by matthee** [comment]
Great! Thank you for your contribution!

**Pull request merged**